pkg/query/querier.go

```diff
@@ -250,9 +256,20 @@ func (q *querier) selectFn(ctx context.Context, hints *storage.SelectHints, ms .
 	if err != nil {
 		return nil, errors.Wrap(err, "convert matchers")
 	}
+	var storeMatchers [][]storepb.LabelMatcher
+	for _, stms := range q.storeMatchers {
+		stm, err := storepb.TranslatePromMatchers(stms...)
```

Comment:
Hm what about translating them in build time (start time of query) not query time? (: